### ISSUE TITLE
msys2 acquiring failure

### ISSUE THREAD

**Issue by jasjuang** [Microsoft/vcpkg#2924]
After removing the folder `vcpkg\downloads\tools\msys2` and `vcpkg\downloads\msys2-base-x86_64-20161025.tar.xz`. `vcpkg install ffmpeg` will show the below error message:

```
-- Acquiring MSYS2...
-- Downloading https://sourceforge.net/projects/msys2/files/Base/x86_64/msys2-base-x86_64-20161025.tar.xz/download...
-- Downloading https://sourceforge.net/projects/msys2/files/Base/x86_64/msys2-base-x86_64-20161025.tar.xz/download... OK
-- Testing integrity of downloaded file...
CMake Error at C:/dev/vcpkg/scripts/cmake/vcpkg_download_distfile.cmake:87 (message):


  File does not have expected hash:

          File path: [ C:/dev/vcpkg/downloads/temp/msys2-base-x86_64-20161025.tar.xz ]
      Expected hash: [ 6c4c18ec59db80b8269698d074866438a624f1ce735ee5005a01b148b02e8f2e966ae381aa1cb4c50f6226c3b7feb271e36907cf26580df084d695b3c9f5c0eb ]
        Actual hash: [ 3b53ccf0243cd4a6a731a898bfdba95c3d22342e7e64f9e9a4a2bce5f74ac809943f570ed92c9077101a178f82f49ece1d1c4bfee1f288ea61d99fb36f6a243e ]

  The file may have been corrupted in transit.  This can be caused by
  proxies.  If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY
  environment variables to
```

The weird thing is it is trying to download `msys2-base-x86_64-20161025.tar.xz` to a temp folder which I think is the root cause of the problem. 

If I go to the link https://sourceforge.net/projects/msys2/files/Base/x86_64/msys2-base-x86_64-20161025.tar.xz/download directly from a browser and download it and manually put it in `vcpkg\downloads`, `vcpkg install ffmpeg` will work without a problem. 

I tried inspecting `vcpkg_acquire_msys.cmake` and `vcpkg_download_distfile.cmake` but couldn't figure out how to solve it. It will be great if someone can fix this.

**Comment by ras0219-msft** [comment]
This is actually due to sourceforge's DOS protection it seems. If you open up the file in a text editor, I get:

```
<html><head>
<title>SourceForge</title>
<!-- <script src="/js/jquery.com/jquery-1.11.0.min.js"></script> -->
<script src="https://code.jquery.com/jquery-1.11.0.min.js"></script>
<script src="https://sourceforge.net/js/mirrors.js"></script>
<script src="/js/sf.js"></script>
<script>
var DR_loc = DR_parse_hash_url();
if (DR_loc) {
    DR_sf_main(DR_loc);
} else {
    window.location.href = 'https://sourceforge.net/home.html';
}
</script>
</head><body>
<noscript>
We're sorry -- the Sourceforge site is currently in Disaster Recovery mode, and currently requires
the use of javascript to function.  Please check back later.
</noscript>
</body></html>
```

**Comment by jasjuang** [comment]
solved by #2941
